ingest/github/collector.py

```
@@ -0,0 +1,427 @@
+
+    while True:
+        results = fetch_fn(page=page)
+        if not isinstance(results, dict):
+            logger.warning(
+                "Pagination fetch returned non-dict result; stopping. type=%s",
+                type(results).__name__,
+            )
+            break
+
+        # GitHub API は配列を直接返す場合と、オブジェクトでラップする場合がある
+        page_items = results if isinstance(results, list) else results
```

Ruff (0.15.1)

[warning] Useless `if`-`else` condition

(RUF034)